### ISSUE TITLE
Sync issue with records falling in/out of scope based on Sharing Settings/Profiles/Perm Sets

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#4205]
**[Mingle Card: 4541](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4541)**
The checksum or orphan/children solution only works for transactional data at this time.
----------------------------------------------------------------------------------------

Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**|4.0.1.3436|
|**Hardware**|Laptop|
|**OS**|Windows|
|**Browser**|Chrome|
|**Username**|
   

tparigi@38demo.com / computer#800 – admin

   

bobfield@38demo.com / computer#700  – field guy

   |
|**Password**| |
|ORG ID| |
|User ID| |
|RayGun Error ID| |
  
 

```

1.  This org currently has the Well object set to Private.  Take a look at [the Sharing Settings for Well](https://na15.salesforce.com/p/own/OrgSharingDetail?setupid=SecuritySharing&st=01Ii00000017fZb) and you will see that no records have been opened up to see for any users besides Admins and those with a profile or Permission Set with _View All _on the Well object.  Bobfield has neither of those characteristics.   
2.  Login to Mobile using bobfield@38demo.com
3.  Open any job and click on the Well picker.  Notice that no wells are avilable.  This is correct.  You can also login to back office using bobifled and click on [the wells tab](https://na15.salesforce.com/a0Q?fcf=00Bi0000003wWTu) and see that no wells are available for this user.
4.  Now open a new Chrome session and login to Back Office with tparigi@38demo.com (an admin) and update [the Sharing Settings for the Well](https://na15.salesforce.com/p/own/OrgSharingDetail?setupid=SecuritySharing&st=01Ii00000017fZb) object to make records available to all Internal Users where the Well’s Available for Field is set to True.  Setup the rule exactly as shown in the following screen shot.
5.  !Sharing.jpg!
6.  Click Save.  Once processing is complete, go back to the bobfield’s [Back Office Well listing](https://na15.salesforce.com/a0Q?fcf=00Bi0000003wWTu) and confirm that he sees several wells now.  He should unless processing is not complete.  
7.  Perform an incremental sync in Mobile with bobfield.  
8.  Open the Well picker on any Job and notice the wells in the picker.

Expected Result
---------------

*   The well picker should include all of the new wells that have come into scope as a result of the Sharing Setting change.  

Actual Result
-------------

*   _No wells are displayed meaning the new wells that have come into scope as a result of sharing settings didn’t sync down to Mobile.  _
*   _The log on the sync page seems to confirm this with “0 new records” listed under Well. _
*   _Note that the reverse of this use case should also work.  In other words, if the user has Wells on their devices that fall OUT of scope due to Sharing Setting changes, those records should fall off of the Mobile database on the next sync.  _

_Analysis_
----------

Related Cards
-------------

Test Plan
---------
